packages/profile-sync-controller/src/shared/encryption/encryption.ts

```diff
@@ -171,6 +171,32 @@ class EncryptorDecryptor {
     return bytesToUtf8(this.#decrypt(ciphertextAndNonce, key));
   }
 
+  getSalt(encryptedDataStr: string) {
+    try {
+      const encryptedData: EncryptedPayload = JSON.parse(encryptedDataStr);
+      if (encryptedData.v === '1') {
+        if (encryptedData.t === 'scrypt') {
+          const { d: base64CiphertextAndNonceAndSalt, saltLen } = encryptedData;
+
+          // Decode the base64.
+          const ciphertextAndNonceAndSalt = base64ToByteArray(
+            base64CiphertextAndNonceAndSalt,
+          );
+
+          // Create buffers of salt and ciphertextAndNonce.
+          const salt = ciphertextAndNonceAndSalt.slice(0, saltLen);
+          return salt;
+        }
+      }
+      throw new Error(
+        `Unsupported encrypted data payload - ${encryptedDataStr}`,
+      );
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : JSON.stringify(e);
+      throw new Error(`Unable to decrypt string - ${errorMessage}`);
+    }
```

Comment:
(you can close this comment, not urgent) Are we sure we want to throw?
Technically we should be safe since the places that use this already to a decrypt first.

Reply:
Could we add a test case I guess?